### PR TITLE
ZUI-79 accordian accessibility

### PR DESCRIPTION
**Pull request by charlielizzy**
Currently component is using span but it should be using the heading of appropriate level for the specific page. 

Changed to make it heading level 3 with styling of h6, with slightly more line spacing for readibility 

Screenshot 2022-12-09 at 15 04 14
Screenshot 2022-12-09 at 15 04 19

### CODE REVIEW

File: src/components/organisms/Accordion/AccordionHeader/__snapshots__/AccordionHeader.test.tsx.snap [hunk 1]
```diff
@@ -2,17 +2,17 @@
...
-  -webkit-letter-spacing: 0;
-  -moz-letter-spacing: 0;
-  -ms-letter-spacing: 0;
-  letter-spacing: 0;
-  color: #2C3236;
   font-size: 16px;
   line-height: 24px;
-  font-weight: 700;
-  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
+  -webkit-letter-spacing: -0.01px;
```
Comment (admmasters) [inline src/components/organisms/Accordion/AccordionHeader/__snapshots__/AccordionHeader.test.tsx.snap:7]: weird that this is affecting the letter spacing

File: src/components/organisms/Accordion/AccordionHeader/AccordionHeader.tsx [hunk 2]
```diff
@@ -76,9 +72,9 @@ const AccordionHeader = ({ children, id, index, textSize = 'body', onClick, ...r
...
     <StyledButton type="button" ref={ref} onClick={handleClick} {...headerPropsRest} {...rest}>
       <TitleContainer>
-        <Title weight="bold" size={textSize}>
+        <Heading as="h3" className="my-0" color={colors.greyDarkest} size="h6">
           {children}
-        </Title>
+        </Heading>
         <Cross active={isActiveSection(index)} />
       </TitleContainer>
     </StyledButton>
```
Comment (moshie) [inline src/components/organisms/Accordion/AccordionHeader/AccordionHeader.tsx:80]: Screenshot 2022-12-12 at 08 56 38

Worth mentioning that adding a heading inside a button is semantically incorrect

Comment (OlenaKashuba) [reply src/components/organisms/Accordion/AccordionHeader/AccordionHeader.tsx:80]: @moshie you are totally right! Girls, we have done it wrong, it should be <h3><button /> </h3> 
image

Comment (charlielizzy) [reply src/components/organisms/Accordion/AccordionHeader/AccordionHeader.tsx:80]: i've made changes to put the button inside the heading now

**Review by moshie — APPROVED** [review]

**Review by poteirard — APPROVED** [review]

**Comment by charlielizzy** [comment]
changes reviewed and approved in https://github.com/zopaUK/react-components/pull/867

**Pull request closed without merging**